testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/messaging/mgmt/PrintDataTestCase.java

```diff
@@ -0,0 +1,296 @@
+
+        // reload in admin-only mode
+        executeReloadAndWaitForCompletion(managementClient, true);
+
+        // export the journal (must be performed in admin-only mode)
+        Path dataPath = new File("print-data.txt").toPath();
+        Files.deleteIfExists(dataPath);
+        printData(dataPath, false, false);
+        String addressId = null;
+        String queueId = null;
+        boolean inMessages = false;
+        boolean hasQueueCount = false;
+        for (String line : Files.readAllLines(dataPath)) {
+            if (addressId == null) {
+                Matcher addressMatcher = ADDRESS_ID.matcher(line);
+                if (addressMatcher.find()) {
+                    addressId = addressMatcher.group(1);
+                }
+            } else {
+                if (queueId == null) {
+                    Matcher queueMatcher = QUEUE_ID.matcher(line);
+                    if (queueMatcher.find()) {
+                        queueId = queueMatcher.group(1);
+                    }
+                } else {
+                    if (!inMessages) {
+                        inMessages = "M E S S A G E S   J O U R N A L".equals(line);
+                    } else {
+                        if (!hasQueueCount) {
+                            hasQueueCount = ("queue id " + queueId + ",count=1").equals(line);
+                        }
+                    }
+                }
+            }
+        }
+        Assert.assertNotNull("Address Id not found", addressId);
+        Assert.assertNotNull("Queue Id not found", queueId);
+        Assert.assertTrue("Should have a message count of 1 for " + jmsQueueName, hasQueueCount);
+        Files.deleteIfExists(dataPath);
+        executeReloadAndWaitForCompletion(managementClient, false);
+    }
+
+    @Test
+    public void testPrintSafeData() throws Exception {
+        // send a persistent message
+        removeAllMessagesFromQueue(jmsQueueName);
+        String text = "print-safe-data";
+        sendMessage(remoteContext, jmsQueueLookup, text);
+        Assert.assertEquals(1, countMessagesInQueue(jmsQueueName));
+
+        // reload in admin-only mode
+        executeReloadAndWaitForCompletion(managementClient, true);
+
+        // export the journal (must be performed in admin-only mode)
+        Path dataPath = new File("print-safe-data.txt").toPath();
+        Files.deleteIfExists(dataPath);
+        printData(dataPath, true, false);
+        boolean inMessages = false;
+        boolean hasQueueCount = false;
+        for (String line : Files.readAllLines(dataPath)) {
+            if (!inMessages) {
+                inMessages = "M E S S A G E S   J O U R N A L".equals(line);
+            } else {
+                if (!hasQueueCount) {
+                    hasQueueCount = SAFE_QUEUE_COUNT.matcher(line).find();
+                }
+            }
+        }
+        Assert.assertTrue("Should have a message count of 1 for " + jmsQueueName, hasQueueCount);
```

Comment:
This check is not enough, the test passes even with `printData(dataPath, false, false); // ~ secure = false` 

but I guess we want to check the secure info is _not printed_ and that is also described in analysis:
> - check that it displays the correct number of total messages and the correct number of message in the queue but doesn't display the queue infos.

Reply:
Updated, thanks.